test/index.js

```diff
@@ -549,6 +548,157 @@ describe("mongoose_delete with options: { deletedBy : true, deletedByType: Strin
+            should.not.exist(success.deletedId);
+        } catch (err) {
+            should.not.exist(err);
+        }
+    });
+});
+
+describe("mongoose_delete with options: { deletedId : true }, using option: typeKey", function () {
+
+    const deletedIdSchema2 = new Schema({name: String}, {collection: 'mongoose_delete_test_deleted_id_2', typeKey: '$type'});
+    deletedIdSchema2.plugin(mongoose_delete, { deletedId: true });
+    const TestDeletedId2 = mongoose.model('TestDeletedId2', deletedIdSchema2);
+
+    let puffy1 = null;
+    let puffy2 = null;
+
+    beforeEach(async function () {
+        const created = await TestDeletedId2.create(
+          [
+              { name: 'Puffy1' },
+              { name: 'Puffy2' },
+              { name: 'Puffy3', deleted: true, deletedId: "53da93b16b4a6670076b16bf" }
+          ]
+        );
+
+        puffy1 = { ...created[0]._doc };
+        puffy2 = { ...created[1]._doc };
+    });
+
+    afterEach(async function () {
+        await mongoose.connection.db.dropCollection("mongoose_delete_test_deleted_id_2");
+    });
+
+    const actionId = getNewObjectId("53da93b16b4a6670076b16bf")
+
+    it("delete() -> should save `deletedId` key", async function () {
+        try {
+            const puffy = await TestDeletedId2.findOne({name: 'Puffy1'});
+            const success = await puffy.delete( {deletedId: actionId} );
+            success.deletedId.toString().should.equal(actionId.toString());
+        } catch (err) {
+            should.not.exist(err);
+        }
+    });
+
+    it("restore() -> should set deleted:false and delete deletedId key", async function () {
+        try {
+            const puffy = await TestDeletedId2.findOne({name: 'Puffy3'});
+            const success = await puffy.restore();
+            success.deleted.should.equal(false);
+            should.not.exist(success.deletedId);
+        } catch (err) {
+            should.not.exist(err);
+        }
+    });
+});
+
+describe("mongoose_delete with options: { deletedId : true, deletedIdType: String }", function () {
+
+    const deletedIdSchema3 = new Schema({name: String}, {collection: 'mongoose_delete_test_deleted_id_3'});
+    deletedIdSchema3.plugin(mongoose_delete, { deletedId: true, deletedIdType: String });
+    const TestDeletedId3 = mongoose.model('TestDeletedId3', deletedIdSchema3);
+
+    let puffy1 = null;
+    let puffy2 = null;
+
+    beforeEach(async function () {
+        const created = await TestDeletedId3.create(
+          [
+              { name: 'Puffy1' },
+              { name: 'Puffy2' },
+              { name: 'Puffy3', deleted: true, deletedBy: "custom_user_id_12345678" }
+          ]
+        );
+
+        puffy1 = { ...created[0]._doc };
+        puffy2 = { ...created[1]._doc };
+    });
+
+    afterEach(async function () {
+        await mongoose.connection.db.dropCollection("mongoose_delete_test_deleted_id_3");
+    });
+
+    const actionIdCustom = "custom_user_id_12345678";
```

Comment:
🤏 Maybe something like
```suggestion
    const actionIdCustom = "delete_asset_group_1234";
```